[Test]Creating TC_MOD_1_2 python module:

j-ororke opened the pull request:
- Creating TC_MOD_1_2 python module on task [#483](https://github.com/project-chip/matter-test-scripts/issues/483)

#### Testing
- WSL
- REPL docker container
- TH {"version": "spring2025", "SHA": "953169b-local"}

j-ororke committed:
Creating TC_MOD_1_2 python module:
- Creating TC_MOD_1_2 python module based on task by Cecille @ https://github.com/project-chip/matter-test-scripts/issues/483

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Updating TC_MOD_1_2 test module:
- Resolving linting errors

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Updated TC_MOD_1_2 python test module:
- Updated to include ref links for info on StandardNamespace and OnMode values

cecille commented on src/python_testing/TC_MOD_1_2.py:
```diff
@@ -0,0 +1,108 @@
+        MOD_cluster = Clusters.ModeSelect
+        attributes = MOD_cluster.Attributes
+
+        if await self.attribute_guard(endpoint=self.endpoint, attribute=attributes.SupportedModes):
```
Supported modes is required, so doesn't need an attribute guard. The PICS marking can be removed from the test.

j-ororke replied:
This has been updated, thank you!

cecille commented on src/python_testing/TC_MOD_1_2.py:
```diff
@@ -0,0 +1,108 @@
+            modes = [option.mode for option in mode_options]
+
+        self.step(2)
+        if await self.attribute_guard(endpoint=self.endpoint, attribute=attributes.CurrentMode):
```
same - mandatory attribute.

j-ororke replied:
This has been updated, thank you!

cecille commented on src/python_testing/TC_MOD_1_2.py:
```diff
@@ -0,0 +1,108 @@
+        attributes = MOD_cluster.Attributes
+
+        if await self.attribute_guard(endpoint=self.endpoint, attribute=attributes.SupportedModes):
+            mode_options = await self.read_single_attribute(dev_ctrl=self.th1, node_id=self.dut_node_id, endpoint=self.endpoint, attribute=attributes.SupportedModes)
```
The checks in the test plan are more in depth here. I don't think we have a machine-readable version of the standard namespaces right now, so leave that piece for now, but there are some checks that should be in here. See test plan. In particular, unique modes.

cecille commented on src/python_testing/TC_MOD_1_2.py:
```diff
@@ -0,0 +1,108 @@
+            asserts.assert_in(Start_Up_Mode_ID, modes, "Current Mode ID should have been in supported modes")
+
+        self.step(5)
+        if await self.attribute_guard(endpoint=self.endpoint, attribute=attributes.Description):
```
same here - description is mandatory

j-ororke replied:
This has been updated, thank you!

cecille commented on src/python_testing/TC_MOD_1_2.py:
```diff
@@ -0,0 +1,108 @@
+            assert isinstance(Description, str), "Description was not a human readable string"
+
+        self.step(6)
+        if await self.attribute_guard(endpoint=self.endpoint, attribute=attributes.StandardNamespace):
```
And here

j-ororke replied:
This has been updated, thank you!

cecille commented on src/python_testing/TC_MOD_1_2.py:
```diff
@@ -0,0 +1,108 @@
+        if await self.attribute_guard(endpoint=self.endpoint, attribute=attributes.StandardNamespace):
+            # Currently appears to return a 0 by default, need to check on this to make sure it is acceptable
+            #Ref Link: https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/06c4d55962954546ecf093c221fe1dab57645028/src/app_clusters/ModeSelect.adoc#62-standardnamespace-attribute
+            Standard_Namespace = await self.read_single_attribute(dev_ctrl=self.th1, node_id=self.dut_node_id, endpoint=self.endpoint, attribute=attributes.StandardNamespace)
```
```suggestion
            standard_namespace = await self.read_single_attribute(dev_ctrl=self.th1, node_id=self.dut_node_id, endpoint=self.endpoint, attribute=attributes.StandardNamespace)
```

cecille replied:
The verification here should be that this is actually a real namespace, which I don't think 0 is. But also, we don't have a machine-readable source yet, so let's handle that in a follow up.

cecille replied:
just talked to hasty, and we have a parser for these. I won't have a chance to do anything about it for a few days, but let me see about grabbing the data. Maybe you can be the person who writes the parser - that's more fun than test cases for the most part.

j-ororke replied:
That sounds great to me, I would enjoy if I was able to be the one who got to write the parser here!

j-ororke replied:
PR for the above action item can be found here: [#4](https://github.com/cecille/connectedhomeip/pull/4)

**Note:** The above is a link to a PR to be merged to Cecille's PR #[37527](https://github.com/project-chip/connectedhomeip/pull/37527) based off the work she has done with the namespace XML's. 
The PR #4 provided above is just the quick unit test and the parser created to complete this action item with the data provided within PR #37527

j-ororke committed:
Apply suggestions from code review

- Updating test code based on recent suggestions from Cecille

Co-authored-by: C Freeman <cecille@google.com>

j-ororke committed:
Updating TC_MOD_1_2 test module:
- Updating test module to remove checks for mandatory attributes

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Updating TC_MOD_1_2 test module:
- Readding missing test step identifiers

j-ororke committed:
Updating TC_MOD_1_2 test module:
- Re-input missing test step numbers

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Updating TC_MOD_1_2 test module:
- Added check to make sure that all values for modes in SupportedModes are unique values

restyled-commits committed:
Restyled by autopep8

j-ororke committed:
Merge branch 'master' into create_MOD_1_2_test
